src/main/webapp/loginScript.js

```diff
@@ -1,10 +1,21 @@
 function onSignIn(googleUser) {
   var profile = googleUser.getBasicProfile();
-  localStorage.setItem("username",profile.getName()); //sends player's names into local storage
-  localStorage.setItem("ID", profile.getId()); //sends player's id into local storage
+  var name = profile.getName();
+  var id = profile.getId();
+  localStorage.setItem("username",name); //sends player's names into local storage
+  localStorage.setItem("ID", id); //sends player's id into local storage
+  updateUserEntity(name,id);
   window.location = "index.html";
 }
 
+//create/update user entity
+function updateUserEntity(name, id){
+    var xhttp = new XMLHttpRequest();
+    xhttp.open("POST", "/user", false);
+    xhttp.setRequestHeader("Content-type", "application/x-www-form-urlencoded");
+    xhttp.send("name=" +name+ "&id=" +id);
+}
+
```

Comment:
are you sure we want to update the user's information in js? Would it be easier to do in a servlet and safer too? Also, how come we would need to update the user's information here? I thought this file just checks if the user is logged in or not

Reply:
i just took out the ID, but the only thing this is updating otherwise is the username in the corner which isn't so bad if the client manipulates it. also I realize the function is named bad, but really what it does is it creates a user entity for everyone logged in if they do not already have one. I changed the name of the function to reflect that a little more

Reply:
its always checking the ID not the name, so having the name be fiddled w shouldnt change much